pyaedt/modeler/Primitives3D.py

```diff
@@ -345,6 +346,70 @@ def create_sphere(self, position, radius, name=None, matname=None):
         new_object_name = self._oeditor.CreateSphere(vArg1, vArg2)
         return self._create_object(new_object_name)
 
+    @aedt_exception_handler
+    def create_torus(self, center, major_radius, minor_radius, axis=None, name=None, material_name=None):
+        """Create a torus.
+
+        Parameters
+        ----------
+        center : list
+            Center point for the torus in a list of ``[x, y, z]`` coordinates.
+        major_radius : float
+           Major radius of the torus.
+        minor_radius : float
+           Minor radius of the torus.
+        axis : str, optional
+            Axis of revolution.
+            The default is ``None``, in which case the Z axis is used.
+        name : str, optional
+            Name of the torus. The default is ``None``, in which case the
+            default name is assigned.
+        material_name : str, optional
+            Name of the material.  The default is ``None``, in which case the
+            default material is assigned. If the material name supplied is
+            invalid, the default material is assigned.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.Object3d.Object3d`
+            3D object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateTorus
+
+        Examples
+        --------
+
+        >>> from pyaedt import Hfss
+        >>> hfss = Hfss()
+        >>> origin = [0,0,0]
+        >>> torus = hfss.modeler.primitives.create_torus(origin, major_radius=1,
+        ...                                             minor_radius=0.5, axis="Z",
+        ...                                             name="mytorus", material_name="copper")
+
+        """
+        assert len(center) == 3, "Center argument must be a valid 3 elements List."
+        assert major_radius > 0 and minor_radius > 0, "Both major and minor radius must be greater than 0."
+        assert minor_radius < major_radius, "Major radius must be greater than minor radius."
+
+        x_center, y_center, z_center = self._pos_with_arg(center)
+        axis = GeometryOperators.cs_axis_str(axis)
+        major_radius = self._arg_with_dim(major_radius)
+        minor_radius = self._arg_with_dim(minor_radius)
+
+        vArg1 = ["NAME:TorusParameters"]
+        vArg1.append("XCenter:="), vArg1.append(x_center)
+        vArg1.append("YCenter:="), vArg1.append(y_center)
+        vArg1.append("ZCenter:="), vArg1.append(z_center)
+        vArg1.append("MajorRadius:="), vArg1.append(major_radius)
+        vArg1.append("MinorRadius:="), vArg1.append(minor_radius)
+        vArg1.append("WhichAxis:="), vArg1.append(axis)
+        vArg2 = self._default_object_attributes(name=name, matname=material_name)
+        new_object_name = _retry_ntimes(10, self._oeditor.CreateTorus, vArg1, vArg2)
```

Comment:
Parameters should not be capitalized.

```suggestion
        varg1 = ["NAME:TorusParameters"]
        varg1.append("XCenter:="), varg1.append(x_center)
        varg1.append("YCenter:="), varg1.append(y_center)
        varg1.append("ZCenter:="), varg1.append(z_center)
        varg1.append("MajorRadius:="), varg1.append(major_radius)
        varg1.append("MinorRadius:="), varg1.append(minor_radius)
        varg1.append("WhichAxis:="), varg1.append(axis)
        varg2 = self._default_object_attributes(name=name, matname=material_name)
        new_object_name = _retry_ntimes(10, self._oeditor.CreateTorus, varg1, varg2)
```